templates/tftmpl/testdata/terraform_cs.tfvars.tmpl

```diff
@@ -0,0 +1,30 @@
+# This file is generated by Consul Terraform Sync.
+#
+# The HCL blocks, arguments, variables, and values are derived from the
+# operator configuration for Sync. Any manual changes to this file
+# may not be preserved and could be overwritten by a subsequent update.
+#
+# Task: test
+# Description: user description for task named 'test'
+
+{{- with $allCatalogServices := services }}
+  {{- range $cs := $allCatalogServices }}
+    {{- if regexMatch ".*" $cs.Name }}{{/* Empty template. Detects changes in catalog-services */}}
```

Comment:
The empty template is an interesting work-around, TIL!

Would this be replaced with the `enable_tf_vars` functionality later or would it using a different template block?

Reply:
Good question! As of now, I was thinking it would be used when `enable_tf_vars=false` and then a different template block for when `enable_tf_vars=true`. Also open to changing this especially after seeing how the hcat changes work to detect catalog-services updates. Let me know if you have any thoughts!

Reply:
Thanks for sharing your plans. I don't know exactly how nesting conditional templating works and if the the `enable_tf_vars` evaluation was a part of this inner portion would add unnecessary overhead for the false case. I'm mostly interested in following along to see how it works and don't have thoughts that lean one way over another.

Reply:
Sounds good! If you have thoughts in the future, don't hesitate to share!